### PR TITLE
chore: bump operator-rs to 0.60.1

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
# Description

Supersedes https://github.com/stackabletech/airflow-operator/pull/364. (I renamed the branch, then Github closed the old PR, oops).

- **BREAKING** Removed legacy node selector on roleGroups.
- Use new label builders.
- Use new ldap::AuthorizationClassProvider `endpoint_url()` method

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: rust/operator-binary/src/airflow_controller.rs [hunk 1]
```diff
@@ -16,11 +16,14 @@ use stackable_airflow_crd::{
...
         resources::ResourceRequirementsBuilder, ConfigMapBuilder, ContainerBuilder,
-        ObjectMetaBuilder, PodBuilder, PodSecurityContextBuilder, VolumeBuilder,
+        ObjectMetaBuilder, ObjectMetaBuilderError, PodBuilder, PodSecurityContextBuilder,
+        VolumeBuilder,
     },
     cluster_resources::{ClusterResourceApplyStrategy, ClusterResources},
     commons::{
-        authentication::{AuthenticationClass, AuthenticationClassProvider},
+        authentication::{
+            ldap::AuthenticationProviderError, AuthenticationClass, AuthenticationClassProvider,
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply rust/operator-binary/src/airflow_controller.rs]: fixed in a5a43a7abcf40fae52cfb9d9851e16ef885b9b65

[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply rust/operator-binary/src/airflow_controller.rs]: fixed in a5a43a7abcf40fae52cfb9d9851e16ef885b9b65

File: rust/operator-binary/src/config.rs [hunk 2]
```diff
@@ -67,12 +67,12 @@ fn append_ldap_config(config: &mut BTreeMap<String, String>, ldap: &LdapAuthenti
...
             "{protocol}{server_hostname}:{server_port}",
-            protocol = match ldap.tls {
+            protocol = match ldap.tls.tls {
                 None => "ldap://",
                 Some(_) => "ldaps://",
             },
             server_hostname = ldap.hostname,
-            server_port = ldap.port.unwrap_or_else(|| ldap.default_port()),
+            server_port = ldap.port(),
         ),
```
Comment (NickLarsenNZ) [inline rust/operator-binary/src/config.rs:76]: @sbernauer I have made the endpoint_url a method, as per https://github.com/stackabletech/operator-rs/pull/705/files, but it is fallable, and this function is not (no return type).

What do you suggest for this PR?
- leave as-is?
- make the return type of this function a `Result<()>`, and fix the callers?

Comment (NickLarsenNZ) [reply rust/operator-binary/src/config.rs:76]: For context, see the discussion in the old PR: https://github.com/stackabletech/airflow-operator/pull/364#discussion_r1436814518

Comment (sbernauer) [reply rust/operator-binary/src/config.rs:76]: Haven't looked at the code, but I would prefer to use the new fn and propagate the error correctly

**Review by sbernauer — APPROVED** [review]
Started test. LGTM when test passes
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/116/

**Comment by sbernauer** [comment]
Retry https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/117/

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/119/

**Review by sbernauer — APPROVED** [review]
Test passed :partying_face:

**Pull request merged**